### PR TITLE
Add Django 4.0 trigram word classes

### LINKED ISSUE #1277: Missing TrigramWordSimilarity, TrigramWordDistance
# Bug report

Hi there, thanks for the great library!

## What's wrong

django-stubs is missing type hints for some of the postgres full text search classes that were added in Django 4.0

https://github.com/django/django/blob/8d160f154f0240a423e83ffe0690e472f837373c/django/contrib/postgres/search.py

- `TrigramWordBase`
- `TrigramWordDistance`
- `TrigramStrictWordDistance`
- `TrigramWordSimilarity`
- `TrigramStrictWordSimilarity`

django-stubs has existing type hints for `TrigramBase`, `TrigramDistance`, etc so it should be pretty straightforward to copy those examples.

## How is that should be

I should be able to import these classes without raising a type error.

## System information

- OS:
- `python` version:
- `django` version:
- `mypy` version:
- `django-stubs` version: 1.13.0
- `django-stubs-ext` version:

### PR DESCRIPTION
**Pull request by michael-lazar**
# I have made things!

Documentation for these classes:

 https://docs.djangoproject.com/en/4.1/ref/contrib/postgres/search/#trigramwordsimilarity

## Related issues

- Closes [#1277](https://github.com/typeddjango/django-stubs/issues/1277)

### CODE REVIEW

File: django-stubs/contrib/postgres/search.pyi [hunk 1]
```diff
@@ -104,9 +104,27 @@ class SearchHeadline(Func):
 class TrigramBase(Func):
     def __init__(self, expression: _Expression, string: str, **extra: Any) -> None: ...
 
+class TrigramWordBase(Func):
+    def __init__(self, expression: _Expression, string: str, **extra: Any) -> None: ...
```
Comment (intgr) [inline django-stubs/contrib/postgres/search.pyi]: Hmm Django documentation has the arguments in the other order, for `TrigramWordSimilarity` https://docs.djangoproject.com/en/4.1/ref/contrib/postgres/search/#django.contrib.postgres.search.TrigramWordSimilarity

```
TrigramWordSimilarity(string, expression, **extra)
```

Or is the documentation wrong?

File: django-stubs/contrib/postgres/search.pyi [hunk 2]
```diff
@@ -104,9 +104,27 @@ class SearchHeadline(Func):
...
+
+class TrigramStrictWordDistance(TrigramWordBase):
+    function: str
+    arg_joiner: str
+
+class TrigramWordSimilarity(TrigramWordBase):
+    function: str
+
+class TrigramStrictWordSimilarity(TrigramWordBase):
+    function: str
```
Comment (intgr) [inline django-stubs/contrib/postgres/search.pyi]: The `Func` base class already defines `function: str`, no need to repeat attributes already defined higher in class hierarchy.

**Review by intgr — CHANGES_REQUESTED** [review]
Thanks for the PR

File: django-stubs/contrib/postgres/search.pyi (hunk 1 above)
Comment (michael-lazar) [reply django-stubs/contrib/postgres/search.pyi]: The documentation is correct and I need to fix this, thanks for double checking!

IMO it's super confusing that the arguments are swapped with the other trigram functions, but the rationale is given in the Django PR: https://github.com/django/django/pull/14833#discussion_r702749182

File: django-stubs/contrib/postgres/search.pyi (hunk 2 above)
Comment (michael-lazar) [reply django-stubs/contrib/postgres/search.pyi]: Sure, I can do the same thing for `arg_joiner` too.

**Review by intgr — APPROVED** [review]
Looks good 👍

**Comment by intgr** [comment]
CI will be fixed in #1279

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks both of you!

**Pull request merged**